bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3109,6 +3092,16 @@ void ReplicaImp::onMessage<ReqMissingDataMsg>(ReqMissingDataMsg *msg) {
   delete msg;
 }
 
+void ReplicaImp::askToLeaveView(ReplicaAsksToLeaveViewMsg::Reason reasonToLeave) {
```

Comment:
can we move this method to the `ViewsManager`, instead of having it as a private method in `ReplicaImp` ?

Reply:
This method was created in order to remove some of the code duplication in `ReplicaImp`. It wraps a call to `tryToGotoNextView()`, which eventually boils down to a call to `MoveToHigherView()` (another member function of `ReplicaImp`). Unfortunately the call to 'MoveToHigherView()' is a major drawback, which prevents a lot of the code from being moved to `ViewsManager`.